src/main/java/org/jabref/logic/pdf/search/retrieval/LuceneSearcher.java

```diff
@@ -0,0 +1,120 @@
+    private final Directory indexDirectory;
+
+    private LuceneSearcher(BibDatabaseContext databaseContext, Directory indexDirectory) {
+        this.databaseContext = databaseContext;
+        this.indexDirectory = indexDirectory;
+    }
+
+    public static LuceneSearcher of(BibDatabaseContext databaseContext) throws IOException {
+        return new LuceneSearcher(databaseContext, new NIOFSDirectory(databaseContext.getFulltextIndexPath()));
+    }
+
+    /**
+     * Search for results matching a query in the Lucene search index
+     *
+     * @param searchString a pattern to search for matching entries in the index, must not be null
+     * @param maxHits      number of maximum search results, must be positive
+     * @return a result set of all documents that have matches in any fields
+     */
+    public LuceneSearchResults search(final String searchString, final int maxHits)
+        throws IOException {
+        Thread.dumpStack();
+        if (StringUtil.isBlank(Objects.requireNonNull(searchString, "The search string was null!"))) {
+            return new LuceneSearchResults();
+        }
+        if (maxHits <= 0) {
+            throw new IllegalArgumentException("Must be called with at least 1 maxHits, was" + maxHits);
+        }
+
+        List<SearchResult> resultDocs = new LinkedList<>();
+
+        if (!DirectoryReader.indexExists(indexDirectory)) {
+            LOGGER.debug("Index directory {} does not yet exist", indexDirectory);
+            return new LuceneSearchResults();
+        }
+
+        try (IndexReader reader = DirectoryReader.open(indexDirectory)) {
+            IndexSearcher searcher = new IndexSearcher(reader);
+            String[] searchable_fields = new String[SearchFieldConstants.searchableBibFields.size()];
+            SearchFieldConstants.searchableBibFields.toArray(searchable_fields);
+            Query query = new MultiFieldQueryParser(searchable_fields, new EnglishStemAnalyzer()).parse(searchString);
+            TopDocs results = searcher.search(query, maxHits);
+            for (ScoreDoc scoreDoc : results.scoreDocs) {
+                resultDocs.add(new SearchResult(searcher, query, scoreDoc));
+            }
+            return new LuceneSearchResults(resultDocs);
+        } catch (ParseException e) {
+            LOGGER.warn("Could not parse query: '{}'!\n{}", searchString, e.getMessage());
+            return new LuceneSearchResults();
+        }
+    }
+
+    public HashMap<BibEntry, List<SearchResult>> search(SearchQuery query) {
+        HashMap<BibEntry, List<SearchResult>> results = new HashMap<>();
+        HashMap<String, Float> boosts = new HashMap<String, Float>();
+        SearchFieldConstants.searchableBibFields.stream().forEach(field -> boosts.put(field, Float.valueOf(4)));
+
+        if (query.getSearchFlags().contains(SearchRules.SearchFlags.FULLTEXT)) {
+            Arrays.stream(SearchFieldConstants.PDF_FIELDS).forEach(field -> boosts.put(field, Float.valueOf(1)));
+        }
```

Comment:
Easy, but ugly. Maybe define a boost property for the standard Fields and get that by Field-name?